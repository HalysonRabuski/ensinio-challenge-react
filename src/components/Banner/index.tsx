import React from 'react'
import { useTranslation } from 'react-i18next'
import Button from '../Button'
import {
  BannerContainer,
  LeftBanner,
  RightBanner,
  ImageCont,
  BgImage,
} from './styles'
import { ReactComponent as Devices } from '../../assets/icons/devices.svg'
import PlaylistBg from '../../assets/icons/icon-playlists.svg'
import CertifiedBg from '../../assets/icons/icon-certified.svg'
import Play from '../../assets/icons/play.png'
import img from '../../assets/images/ISTOCKPHOTO.png'

const Banner: React.FC = () => {
  const { t } = useTranslation()

  return (
    <BannerContainer data-testid="banner-id">
      <BgImage src={PlaylistBg} left="5%" top="5%" />
      <BgImage src={CertifiedBg} left="50%" top="10%" />
      <BgImage src={PlaylistBg} right="5%" top="5%" />
      <LeftBanner>
        <div className="content">
          <p>
            <i>
              <Devices />
            </i>
            {t('banner.platform')}
          </p>
          <h1>{t('banner.title')}</h1>
          <p>{t('banner.description')}</p>
          <div className="actions">
            <Button
              padding="18px 20px"
              bgColor="#00E1E7"
              color="#130c25"
              border="none"
            >
              <b>{t('banner.getstarted')}</b>
            </Button>
            <a id="video" href="/#">
              <i id="play-icon">
                <img src={Play} alt="play-icon" />
              </i>
              {t('banner.video')}
            </a>
          </div>
        </div>
      </LeftBanner>
      <RightBanner>
        <div className="content">
          <div className="circle" />
          <ImageCont src={img} />
        </div>
      </RightBanner>
    </BannerContainer>
  )
}

export default Banner
