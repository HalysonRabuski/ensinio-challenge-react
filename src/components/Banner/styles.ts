import styled from 'styled-components'

interface BgImageProps {
  top?: string
  left?: string
  right?: string
  bottom?: string
}

export const BannerContainer = styled.div`
  background: linear-gradient(259.95deg, #41b5d9 2.25%, #5f41d9 100.27%);
  height: 80vh;
  display: flex;

  @media screen and (min-width: 0px) and (max-width: 480px) {
    display: block;
    height: auto;
    width: auto;
    padding-top: 50%;

    text-align: center;
  }

  @media screen and (min-width: 481px) and (max-width: 1024px) {
    display: block;
    height: auto;
    width: auto;
    padding-top: 20%;

    text-align: center;
  }
`

export const LeftBanner = styled.div`
  width: 50vw;
  margin: auto;
  /* background-color: red; */

  .content {
    margin: 0 auto;
    width: 42%;

    .actions {
      display: flex;
      align-items: center;

      a#video {
        display: flex;
        align-items: center;
        margin-left: 40px;

        i {
          margin-right: 5px;
        }
      }
    }
  }

  p {
    display: flex;
    margin-bottom: 35px;

    i {
      margin-right: 5px;
    }
  }
  h1 {
    margin-bottom: 35px;
  }

  .actions {
    height: 100%;
    padding-top: 20px;
  }

  @media screen and (min-width: 0px) and (max-width: 480px) {
    /* margin-top: 10%; */
    .actions {
      margin: 0 auto;
      text-align: center;
      flex-direction: column;
      justify-content: center;
    }

    text-align: center;

    a#video {
      margin-top: 25px;
      margin-left: 0 !important;
    }

    .content {
      width: 80%;
      text-align: center;
    }

    p {
      justify-content: center;
    }

    display: block;
    height: auto;
    width: auto;
  }
  @media screen and (min-width: 481px) and (max-width: 1024px) {
    .actions {
      margin: 0 auto;
      text-align: center;
      flex-direction: column;
      justify-content: center;
    }
    text-align: center;

    a#video {
      margin-top: 25px;
      margin-left: 0 !important;
    }

    .content {
      width: 80%;
      text-align: center;
    }

    p {
      justify-content: center;
    }

    display: block;
    height: auto;
    width: auto;
  }

  @media screen and (min-width: 1025px) and (max-width: 1430px) {
    .actions {
      margin: 0 auto;
      justify-content: center;
    }
    /* text-align: center; */

    .content {
      width: 80%;
      text-align: center;
    }

    p {
      justify-content: center;
    }
  }
`
export const RightBanner = styled.div`
  width: 50vw;
  /* height: 100%; */
  margin-top: auto;
  .content {
    margin-top: auto;
    height: 100%;

    overflow: hidden;
  }

  .circle {
    z-index: 1;
    position: relative;
    border-radius: 50%;
    background: linear-gradient(
      93.59deg,
      #41b5d9 -79.31%,
      rgba(65, 181, 217, 0) 203.45%
    );
    width: 650px;
    height: 650px;
    transform: translateY(20%);
    opacity: 0.2;
  }

  @media screen and (min-width: 0px) and (max-width: 480px) {
    /* margin-top: 10%; */
    .circle {
      width: 300px;
      height: 300px;
      margin: 0 auto;
    }

    width: 100vw;
    .content {
      width: 100%;
    }
  }

  @media screen and (min-width: 481px) and (max-width: 1024px) {
    .circle {
      margin: 0 auto;
    }

    width: 100vw;
    .content {
      width: 100%;
    }
  }
  @media screen and (min-width: 1025px) and (max-width: 1430px) {
    .circle {
      margin: 0 auto;
      width: 500px;
      height: 500px;
    }

    .content {
      width: 100%;
    }
  }
`
export const ImageCont = styled.img`
  /* object-fit: cover; */
  transform: translateY(-100%) translateX(15%);
  position: absolute;
  z-index: 2;
  /* width: 650px; */

  @media screen and (min-width: 0px) and (max-width: 480px) {
    width: 60%;
    transform: translateY(-100%) translateX(-50%);
  }

  @media screen and (min-width: 481px) and (max-width: 1024px) {
    width: 60%;
    transform: translateY(-100%) translateX(-50%);
  }
  @media screen and (min-width: 1025px) and (max-width: 1430px) {
    width: 40%;
    transform: translateY(-100%) translateX(20%);
  }
`

export const BgImage = styled.img<BgImageProps>`
  position: absolute;
  top: ${(props) => props.top};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  bottom: ${(props) => props.bottom};
  /* object-fit: cover; */
  /* transform: translateY(-100%) translateX(15%);
  position: absolute;
  z-index: 2; */
  /* width: 650px; */
`
