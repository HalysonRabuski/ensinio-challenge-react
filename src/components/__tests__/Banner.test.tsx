/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { render, screen, cleanup, waitFor } from '@testing-library/react'
import renderer from 'react-test-renderer'
// import '../../i18n'
// import renderer from 'react-test-renderer'
import Banner from '../Banner'

afterEach(() => {
  cleanup()
})

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => ({
    t: (str: any) => str,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
    },
  }),
}))

test('should render card component', async () => {
  render(<Banner />)
  const bannerElement = await waitFor(() => screen.getByTestId('banner-id'))

  expect(bannerElement).toBeInTheDocument()
})

test('should render card component with an image and texts as children', () => {
  const cardElement = renderer.create(<Banner />).toJSON()
  expect(cardElement).toMatchSnapshot()
})
