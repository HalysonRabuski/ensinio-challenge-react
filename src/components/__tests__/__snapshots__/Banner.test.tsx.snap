// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card component with an image and texts as children 1`] = `
<div
  className="sc-gtsrHT gokNrE"
  data-testid="banner-id"
>
  <img
    className="sc-jSFjdj bCKpVP"
    src="icon-playlists.svg"
  />
  <img
    className="sc-jSFjdj hUUYXT"
    src="icon-certified.svg"
  />
  <img
    className="sc-jSFjdj dQZAET"
    src="icon-playlists.svg"
  />
  <div
    className="sc-dlnjwi fqJsdG"
  >
    <div
      className="content"
    >
      <p>
        <i>
          <svg>
            devices.svg
          </svg>
        </i>
        banner.platform
      </p>
      <h1>
        banner.title
      </h1>
      <p>
        banner.description
      </p>
      <div
        className="actions"
      >
        <a
          className="sc-bdnxRM jUznVJ"
          color="#130c25"
          data-testid="button-id"
          width="160px"
        >
          <b>
            banner.getstarted
          </b>
        </a>
        <a
          href="/#"
          id="video"
        >
          <i
            id="play-icon"
          >
            <img
              alt="play-icon"
              src="play.png"
            />
          </i>
          banner.video
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-hKFxyN dfxEbf"
  >
    <div
      className="content"
    >
      <div
        className="circle"
      />
      <img
        className="sc-eCApnc hMuNWu"
        src="ISTOCKPHOTO.png"
      />
    </div>
  </div>
</div>
`;
